File: chwazifake_android/app/src/main/java/com/hkwg/chwazifake/data/repository/CircleRepositoryImpl.kt
package com.hkwg.chwazifake.data.repository

import com.hkwg.chwazifake.domain.model.CircleState
import com.hkwg.chwazifake.domain.repository.CircleRepository
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class CircleRepositoryImpl @Inject constructor() : CircleRepository {
    private val circles = mutableListOf<CircleState>()

    override fun addCircle(circle: CircleState) {
        circles.add(circle)
    }

    override fun removeCircle(circle: CircleState) {
        circles.removeIf { it.position == circle.position }
    }

    override fun updateCircle(circle: CircleState) {
        val index = circles.indexOfFirst { it.position == circle.position }
        if (index != -1) {
            circles[index] = circle
        }
    }

    override fun getCircles(): List<CircleState> = circles.toList()
} 

File: chwazifake_flutter/lib/data/repository/circle_repository_impl.dart
import 'package:chwazifake_flutter/domain/model/circle_state.dart';
import 'package:chwazifake_flutter/domain/repository/circle_repository.dart';

class CircleRepositoryImpl implements CircleRepository {
  final List<CircleState> _circles = [];

  @override
  List<CircleState> getCircles() => List.unmodifiable(_circles);

  @override
  void addCircle(CircleState circle) {
    if (_circles.length < 10) {
      _circles.add(circle);
    }
  }

  @override
  void removeCircle(CircleState circle) {
    _circles.removeWhere((element) => element.id == circle.id);
  }

  @override
  void updateCircle(CircleState circle) {
    final index = _circles.indexWhere((element) => element.id == circle.id);
    if (index != -1) {
      _circles[index] = circle;
    }
  }
} 